aiidalab_widgets_base/structures.py

```diff
@@ -393,20 +401,43 @@ def _validate_and_fix_ase_cell(self, ase_structure, vacuum_ang=10.0):
 
     def _on_file_upload(self, change=None):
         """When file upload button is pressed."""
+        self.output.value = ""
         for fname, item in change["new"].items():
-            frmt = fname.split(".")[-1]
-            if frmt == "cif":
-                self.structure = CifData(file=io.BytesIO(item["content"]))
-            else:
-                with tempfile.NamedTemporaryFile(suffix=f".{frmt}") as temp_file:
-                    temp_file.write(item["content"])
-                    temp_file.flush()
-                    self.structure = self._validate_and_fix_ase_cell(
-                        get_ase_from_file(temp_file.name)
-                    )
+            self.structure = self._read_structure(fname, item["content"])
             self.file_upload.value.clear()
             break
 
+    def _read_structure(self, fname, content):
+        suffix = "".join(pathlib.Path(fname).suffixes)
+        if suffix == ".cif":
+            from CifFile import StarError
```

Comment:
Do you mean just catching an `Exception`? I think it is not suggested.

Reply:
Yeah, I essentially meant a naked except
```python
try:
   ...
except:
   ...
```
which I guess is mostly equivalent to catching the base exception class. Catching a specific exception is nicer, but I don't quite like suddenly importing something from what seems to be a transitive dependency.

Reply:
@csadorf what do you think about this?

Reply:
I agree with @danielhollas that we should not use this specific exception class since it is declared in a transitive dependency.

Catching a more specific exception class here is very difficult, because aiida does not explicitly define which Exceptions can be raised so there is no explicit API contract regarding this.

It seems to me that the errors thrown by pycifrw (transitive) are indeed `StarError` which is derived from `Exception` (should probably be a `Runtime` or `ValueError`). The alternative ways to load Cifs via ase (also transitive) can result in a variety of different exceptions which all appear to be either `Value` or `RuntimeError`s.

In conclusion, I think we have to catch `Exception` here, but we should not use a bare `except:` clause. The reason for this is explained [here](https://stackoverflow.com/a/54948581).